include/zephyr/net/net_if.h

```diff
@@ -1672,7 +1678,25 @@ uint8_t net_if_ipv6_get_hop_limit(struct net_if *iface);
  * @param iface Network interface
  * @param hop_limit New hop limit
  */
-void net_ipv6_set_hop_limit(struct net_if *iface, uint8_t hop_limit);
+void net_if_ipv6_set_hop_limit(struct net_if *iface, uint8_t hop_limit);
```

Comment:
This is a public API change, I think you should leave `net_ipv6_set_hop_limit` in, simply calling `net_if_ipv6_set_hop_limit` and add a deprecation warning.

Reply:
Good point, will fix.